packages/shared/src/components/RankProgress.tsx

```diff
@@ -23,6 +24,63 @@ import styles from './RankProgress.module.css';
 
 const notificationDuration = 300;
 
+interface GetRankElementProps {
+  shownRank: number;
+  rank: number;
+  badgeRef: MutableRefObject<SVGSVGElement>;
+  plusRef: MutableRefObject<HTMLElement>;
+  smallVersion: boolean;
+  showRankAnimation: boolean;
+  animatingProgress: boolean;
+}
+
+type RankElementProps = Pick<
+  GetRankElementProps,
+  'shownRank' | 'rank' | 'badgeRef'
+>;
+const RankElement = ({
+  shownRank,
+  rank,
+  badgeRef,
+}: RankElementProps): ReactElement => (
+  <Rank
+    rank={shownRank}
+    className={classNames(
+      'absolute inset-0 w-2/3 h-2/3 m-auto',
+      styles.rank,
+      shownRank && styles.hasRank,
+    )}
+    ref={badgeRef}
+    colorByRank={rank > 0}
+  />
+);
+
+const getRankElement = ({
+  shownRank,
+  rank,
+  badgeRef,
+  plusRef,
+  smallVersion,
+  showRankAnimation,
+  animatingProgress,
+}: GetRankElementProps): ReactElement => {
+  if (!smallVersion)
+    return (
+      <RankElement shownRank={shownRank} rank={rank} badgeRef={badgeRef} />
+    );
+
+  return !showRankAnimation || !animatingProgress ? (
```

Comment:
It looks like line 69 and line 73 are the same. Maybe we can remove the tertiary operator and merge its condition into the if statement in line 67

Reply:
@idoshamun Would you mind checking if this is what you meant?